apps/worker/app/routers/runs.py

```
@@ -0,0 +1,157 @@
+            # Mark run as failed if session URL is missing
+            await run_service.update_run_status(run_id, RunStatus.FAILED)
+            await emit_progress(
+                run_id,
+                {
+                    "status": RunStatus.FAILED.value,
+                    "message": "Session created without viewer URL",
+                },
+            )
+            raise HTTPException(
+                status_code=502, detail="Session created without viewer URL"
+            )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)